Admin, error: failed to get oidc user info, error: <QuerySeter> no row found

If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
Getting frequent error on the harbor core container 

**Steps to reproduce the problem:**
Please provide the steps to reproduce this problem.
Enable OIDC authentication
enable OIDC user login except admin user



**Versions:**
Please specify the versions of following systems.

- harbor version: 2.11.1

Error Logs:
`2024-09-19T16:56:12Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="242bf2db33e997bf2c69f93d81f51db8"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
2024-09-19T16:56:42Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="fa0b94915681af7a535e11c2af207b0e"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
2024-09-19T16:57:12Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="195ee11d4bd8ea0c39368145cf76c153"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found
2024-09-19T16:57:42Z [ERROR] [/server/middleware/security/oidc_cli.go:68][requestID="4539600fe12d22a94fd11623237bded8"]: failed to verify secret, username: admin, error: failed to get oidc user info, error: <QuerySeter> no row found`

--- a/src/pkg/oidc/dao/meta.go
+++ b/src/pkg/oidc/dao/meta.go
@@ -16,6 +16,7 @@ package dao
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/goharbor/harbor/src/common/models"
 	"github.com/goharbor/harbor/src/lib/errors"
@@ -65,6 +66,9 @@ func (md *metaDAO) GetByUsername(ctx context.Context, username string) (*models.
 	}
 	res := &models.OIDCUser{}
 	if err := ormer.Raw(sql, username).QueryRow(res); err != nil {
+		if errors.Is(err, orm.ErrNoRows) {
+			return nil, fmt.Errorf("oidc user data with username %s not found", username)
+		}
 		return nil, err
 	}
 	return res, nil

--- a/src/server/middleware/security/oidc_cli.go
+++ b/src/server/middleware/security/oidc_cli.go
@@ -63,16 +63,20 @@ func (o *oidcCli) Generate(req *http.Request) security.Context {
 		return nil
 	}
 
-	info, err := oidc.VerifySecret(ctx, username, secret)
-	if err != nil {
-		logger.Errorf("failed to verify secret, username: %s, error: %v", username, err)
-		return nil
-	}
 	u, err := uctl.GetByName(ctx, username)
 	if err != nil {
 		logger.Errorf("failed to get user model, username: %s, error: %v", username, err)
 		return nil
 	}
+
+	info, err := oidc.VerifySecret(ctx, username, secret)
+	if err != nil {
+		if u.UserID != 1 { // skip the admin user
+			logger.Errorf("failed to verify secret, username: %s, error: %v", username, err)
+		}
+		return nil
+	}
+
 	oidc.InjectGroupsToUser(info, u)
 	logger.Debugf("an OIDC CLI security context generated for request %s %s", req.Method, req.URL.Path)
 	return local.NewSecurityContext(u)

